### PR TITLE
Upgrade Tool

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Adds the upgrade tool

### Why is it needed?

To facilitate upgrading Strapi projects by automatically updating version numbers, installing packages, and running codemods for breaking changes.

### How to test it?

Within a Strapi project, run the upgrade tool command (located at packages/utils/upgrade/bin/upgrade.js )

So from examples/getstarted you would run: `../../packages/utils/upgrade/bin/upgrade.js --help` to see the list of commands

Note: The actual code transforms will be added throughout the development of v5 (for example, [this](https://github.com/strapi/strapi/pull/19155)), so at the moment running this tool doesn't cause any effects. So to really test it you will need to add some codemods to the resources dir.

### CODE REVIEW

**Review by Bassel17 — APPROVED** [review]
LGTM

File: packages/utils/upgrade/README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+`../../packages/utils/upgrade/bin/upgrade`
+
+Run the command with the `--help` option to see all the available options.
+
+[Coming Soon] The Strapi Upgrade tool will be available using `npx @strapi/upgrade` and an alias for that within a project using `strapi upgrade`
+
+## Writing a code transforms
+
+To begin your code transform script, create a file `upgrade/resources/transforms/{X.X.X}/{short-description-of-action}.{code|json}.ts` where `X.X.X` is the target version of Strapi the codemod will be run for.
+
```
Comment (Convly) [inline packages/utils/upgrade/README.md:45]:
```suggestion
For example, all breaking changes for the initial release of Strapi v5 will go in upgrade/resources/codemods/5.0.0
```

**Review by Convly — APPROVED** [review]

**Comment by Convly** [comment]
The commitlint is still broken, we might want to fix that before merging?

**Comment by innerdvations** [comment]
> The commitlint is still broken, we might want to fix that before merging?

It can't be fixed directly (afaik, can it?), but if we squash and merge only the new message will be included.

**Comment by Convly** [comment]
We could've rebased + modified the commits names
Hope it won't break on main.

**Comment by innerdvations** [comment]
It shouldn't be a problem, there have been several other v5 merges with the same issue. I think we'll have to tweak the commitlint and CI at some point though and set up some processes to avoid this, like commit hooks so people can't cause the problem to begin with.

**Pull request merged**